vignettes/more-recipes.Rmd

```diff
@@ -0,0 +1,277 @@
+
+```{r dummy-variable}
+length(unique(x$geo_value))
+
+dummies <- r %>% 
+  step_dummy(geo_value) %>%
+  prep(training = x) 
+
+dummy_data <- bake(dummies, x)
+
+names(dummy_data)
+```
+
+Once the pre-processing steps are done, we can move on to model fitting using 
+the `parsnip` package. 
+
+
+# Model Fitting and Post-processing
+
+Sometimes, the exact values are desired, therefore fitting a linear regression 
+or poisson regression would be a good idea. 
+
+On the other hand, say instead of predicting 7-day ahead death counts, we would 
+like to classify if the future predictions are high or low based on some given threshold. 
+In this case, a logistic regression among other classification models can be used.
+For the purpose of illustration, we define two categories: less than 10K(low) 
+and more than 10K(high). 
+
+We resort to the `recipes` package again to change the numerical outcomes to 
+categorical for the purpose of classification:
+
+```{r, warning = FALSE}
+binner <- function(x) {
+  x <- cut(x, breaks = c(0, 10000, Inf), include.lowest = TRUE)
+  # now return the group number
+  as.numeric(x)
+}
+
+inc <- c("low", "high")
+
+r_categorical <- r %>% 
+  step_num2factor(ahead_7_deaths, 
+                  role = "outcome", 
+                  transform = binner, 
+                  levels = inc)
+
+```
+
+As a sanity check we can see now that the preprocessed data has a categorical 
+outcome in `ahead_7_deaths`. 
+
+```{r, warning FALSE, echo = FALSE}
+glimpse(slice_sample(bake(prep(r_categorical, x),x), n = 6))
+```
+
+
+### Linear Regression
+
+First, we can try fitting a linear regression:
+
+```{r linear-fit-workflow, warning = FALSE}
+wf <- epi_workflow(r_lin, parsnip::linear_reg()) %>%
+    fit(x) 
+
+latest <- get_test_data(recipe = r_lin, x = x)
+```
+
+We take a look at the linear regression fit and the values of the coefficients.
+
+```{r linear-fit, warning = FALSE}
+wf$fit$fit # look at the model fit and coefficients
+
+p <- predict(wf, latest) %>% filter(!is.na(.pred))
+
+p
+```
+
+The predictions are saved in the `.pred` column and represents the log-transformed 
+predicted death counts 7 days from Dec 31. 2021. 
+
+
+### Poisson regression
+
+For count data, we can use poisson regression. Although the counts here are not 
+integer since they were calculated from case rates multiplying population, some 
+extra preprocessing steps can help us round them to integers to allow fitting a 
+poisson regression.
+
+
+```{r poisson-fit}
+r <- r %>% 
+  step_mutate(ahead_7_deaths = round(ahead_7_deaths, 0), 
+                       role = "predictor") %>%
+  step_filter(ahead_7_deaths > 0)
+
+latest <- get_test_data(recipe = r, x = x)
+
+wf <- epi_workflow(r, parsnip::poisson_reg()) %>%
+    fit(x) 
+
+predict(wf, latest) %>% filter(!is.na(.pred))
+```
+
+Let's also take a look at the poisson regression fit:
+```{r, echo =FALSE}
+wf$fit$fit
+```
+
+### Logistic Regression
```

Comment:
https://bccovid-19group.ca/publication/mcdonald-2021-12-13-auxiliary/

Use the hotspot prediction model. Three categories, upswing, flat, decrease.

Reply:
still need to add mathematical expression